runtime/src/accounts_db.rs

```diff
@@ -2140,23 +2133,24 @@ impl AccountsDB {
             if let Some(expected_slot) = expected_slot {
                 assert_eq!(*slot, expected_slot);
             }
-            if let Some(slot_storage) = storage.0.get(slot) {
-                if let Some(store) = slot_storage.get(&account_info.store_id) {
-                    assert_eq!(
-                        *slot, store.slot,
-                        "AccountDB::accounts_index corrupted. Storage should only point to one slot"
-                    );
-                    let count = store.remove_account();
-                    if count == 0 {
-                        dead_slots.insert(*slot);
-                    }
+            if let Some(store) = self
+                .storage
+                .get_account_storage_entry(*slot, account_info.store_id)
+            {
+                assert_eq!(
+                    *slot, store.slot,
+                    "AccountDB::accounts_index corrupted. Storage should only point to one slot"
```

Comment:
nits: not this pr specific changes, but how about printing `*slot` in `store.slot` in the custom message? this would be crutial clue for debugging this, if this occurs ever after this pr...  ;)

Reply:
yup, definitely would be helpful, added!